### ISSUE TITLE
Plan out future agility.

### ISSUE THREAD

**Issue by mikewest** [WICG/signature-based-sri#37]
@yoavweiss asked good questions about future evolution of this spec, either adding new algorithms or parameters beyond those currently supported in the profile we've defined. We should add a section to the document sketching out the path we'd expect to take.

**Comment by martinthomson** [comment]
I don't see any significant problem here with agility.  A new scheme will need a new label, but everything flow from there: from the SRI scheme and public key to the signature construction.

My understanding is that you can continue to avoid the "alg" parameter because the integrity attribute establishes the right expectation about the choice of algorithm.  An adversary cannot supply you with an Ed25520[^1] signature in place of your genuinely secure Ed25519.

[^1]: ed25519 public keys are only 255 bits long anyway.  So we already deal with numbers that include fixed zero bits.  Ed25520 keys would be the same size, but over a non-prime field, which might not work out :)  The way you deal with that is to use `Math.floor((n+7)/8)` bytes and zero padding, not any fancy encoding scheme.

**Comment by mikewest** [comment]
I agree with you that shifting the `type` (a la #34) is likely to be the simplest way of addressing algorithmic changes. So we're back to naming... :)

**Comment by mikewest** [comment]
We've dealt with this by allowing (and ignoring) arbitrary parameters. Closing out the issue.
